### PR TITLE
Channel creation, use and connection setups

### PR DESCRIPTION
**Pull request by dxwc**

### CODE REVIEW

**Comment by jiajern** [comment]
ok

File: controller/channel.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+router.get('/channel/:id', (req, res) =>
+{
+    if
+    (
+        val.isUUID(req.params.id, 4) &&
+        global.channels.hasOwnProperty(req.params.id)
+    )
+    {
+        let the_channel = global.channels[req.params.id];
```
Comment (Sevege) [inline controller/channel.js]: what is global?

**Comment by Sevege** [comment]
ok

File: controller/channel.js (hunk 1 above)
Comment (dxwc) [reply controller/channel.js]: Using the global variable to save cache of each of channel's data and all users connected to each channel within it. This is to not constantly have to query database for each connection and only query database when video starts or ends, channel created or removed, or if any data is missing.

**Pull request merged**